Telemetry: Adding in tests for Java Agent 1.29

jakub-pomykala opened the pull request:
- Telemetry with Java Agent 1.29 for MicroProfile 5 and MicroProfile 6.1.
- MicroProfile 6.0 uses Java Agent 1.19
- MicroProfile 1.4 and 4.1 are currently not working with Java Agent 1.29 due to a bug in the Java Agent

This PR does not fix #26919 but is a half way point.

See https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10159

Azquelt committed:
Telemetry: Repeat AgentTest with the v1.29 agent

So far this doesn't work because the test needs adjusting for the
different spans created by the 1.29 agent. In particular JAX-RS requests
appear to get one server span with the request info and one internal
span with the class and method info.

jakub-pomykala committed:
Bisecting where WithSpan stops working on EE7/8

- Work in progress
- Adding in tests for javaAgent.1.29
- Added in new repeatRules for either javaagent.1.19 and 1.29
- build.gradle current has javaagent.1.19 set to 1.24 for testing purposes

jakub-pomykala committed:
Update Javaagent 1.29 tests with withSpan

jakub-pomykala committed:
Fixing javaagent and withspan annotation issue

jakub-pomykala committed:
Updating JavaAgent from 1.19 to 1.29

- Updating JavaAgent to version 1.29
- Correcting the tests pass as expected for all the repeats
- Making EE7 and EE8 pass with Java Agent 1.19 as there is still an on going issue with 1.29 and EE7 and EE8 repeats
- Removing AutoConfiguredOpenTelemetrySdkBuilder.java

- Creating a new test suite for javaagent 1.29 tests
- Updating AgentConfig and AgentConfigMultiApp tests
- Implementing skipForRepeats for 1.29 for tests that are failing due to javaagent 1.29 (e.g. annotated with withSpan)

jakub-pomykala committed:
Updating the FatSuite

Yas-Franco commented on dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/Agent129Test.java:
```diff
@@ -0,0 +1,446 @@
+/*******************************************************************************
+ * Copyright (c) 2022,2023 IBM Corporation and others.
```
Copyright headers need updating to 2024

Yas-Franco commented on dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigMultiAppTest.java:
Copyright header needs updating

Yas-Franco commented on dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentConfigTest.java:
```diff
@@ -183,14 +216,27 @@ public void testDisableInstrumentation() throws Exception {
 
     @Test
+    /*
+     * Skipping for 1.4 and 4.1 as JavaAgent 1.29 currently will not return a span for methods annotated with @withSpan (BUG)
```
Link issue in this comment

Yas-Franco commented on dev/io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat/fat/src/io/openliberty/microprofile/telemetry/internal/tests/AgentTest.java:
Update comment to say you are testing the java agent 1.19 only

Yas-Franco reviewed (commented):
Minor changes needed but otherwise it looks good!

jakub-pomykala committed:
Updating copyright tags with 2024

- adding in link for java agent issue to comment
- updating comments with java agent versions being tested

yasmin-aumeeruddy commented:
#libby
